packages/libs/react-ui/src/components/Form/ComboBox/ComboBox.tsx

```diff
@@ -0,0 +1,88 @@
+import { Button } from '@components/Button/NewButton';
+import type { IInputProps, ITextFieldProps } from '@components/Form';
+import { TextField } from '@components/Form';
+import { ListBox } from '@components/ListBox';
+import { Popover } from '@components/Popover';
+import type { CSSProperties } from 'react';
+import React, { useRef } from 'react';
+import type { AriaComboBoxProps } from 'react-aria';
+import { mergeProps, useComboBox, useFilter } from 'react-aria';
+import { useComboBoxState } from 'react-stately';
+
+export interface IComboBoxProps<T>
+  extends Omit<AriaComboBoxProps<T>, 'id' | 'label'> {
+  id: IInputProps['id'];
+  label: ITextFieldProps['label'];
+  width: CSSProperties['width'];
+  startIcon?: ITextFieldProps['startIcon'];
+}
+
+/**
+ * @see https://react-spectrum.adobe.com/react-aria/useComboBox.html
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const ComboBox = <T extends object>({
+  id,
+  label,
+  width,
+  startIcon,
+  ...props
+}: IComboBoxProps<T>) => {
+  // Setup filter function and state.
+  const { contains } = useFilter({ sensitivity: 'base' });
+  const state = useComboBoxState({ ...props, defaultFilter: contains });
+
+  // Setup refs and get props for child elements.
+  const buttonRef = useRef(null);
+  const inputRef = useRef(null);
+  const listBoxRef = useRef(null);
+  const popoverRef = useRef(null);
+
+  const { buttonProps, inputProps, listBoxProps } = useComboBox(
+    {
+      ...props,
+      inputRef,
+      buttonRef,
+      listBoxRef,
+      popoverRef,
+    },
+    state,
+  );
+
+  return (
+    <div style={{ display: 'inline-flex', flexDirection: 'column' }}>
+      <div style={{ width }}>
+        {/* @ts-expect-error */}
+        <TextField
+          {...inputProps}
+          label={label}
+          id={id}
+          startIcon={startIcon}
+          ref={inputRef}
+        >
+          <Button {...buttonProps} ref={buttonRef}>
```

Comment:
could you use the Button from NewButton instead, it also has a text variant which can fit more in this use case

Reply:
I did use the Icon now, but not the text variant yet. Do you think that would still indicate enough that it's a clickable element that can/will open the options list?